airbyte/caches/_state_backend.py

```diff
@@ -31,7 +31,8 @@
     from airbyte._future_cdk.state_providers import StateProviderBase
 
 
-STATE_TABLE_NAME = "_airbyte_state"
+CACHE_STATE_TABLE_NAME = "_airbyte_state"
+DESTINATION_STATE_TABLE_NAME = "_airbyte_destination_state"
```

Comment:
Adds a new (third) internal table: `_airbyte_destination_state`. In contrast with `_airbyte_state`, this new table handles state for _external_ destinations and not the internal cache.